adobecom/milo
# MWPW-145574 - Dynamic Nav

- Adds code to add the gnav source to sessionStorage to allow for the same nav experience in subsequent clicks
- `entry` pages will set gnavSource to sessionStorage is a gnav source meta tag is present
- `on` pages will check for a gnavSource in sessionStorage, and if one is found, either modify the gnav-source meta tag, or create one
- pages without the dynamic-nav metadata will do nothing or remove gnavSource from sessionStorage

Context:
The wiki in the ticket provides more context, but it was desired by our stakeholders that this only be a single tab experience.

Resolves: [MWPW-145574](https://jira.corp.adobe.com/browse/MWPW-145574)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/slavin/dynamic-nav/hugo-boss-case-study-entry?martech=off
- After: https://dynamic-nav--milo--jasonhowellslavin.hlx.page/drafts/slavin/dynamic-nav/hugo-boss-case-study-entry?martech=off
- Before: https://main--bacom--adobecom.hlx.page/drafts/slavin/dynamic-nav/aem-rtcdp-entry?martech=off
- After: https://dynamic-nav-config--bacom--adobecom.hlx.page/drafts/slavin/dynamic-nav/aem-rtcdp-entry?milolibs=dynamic-nav&martech=off
To test: 

1. Go to the second after link. On the Hugo Boss page
2. Observe the nav (pricing and resources)
3. Right click "watch now" and open in a new tab
4. Observe the nav is the default milo nav ("Explore")
5. From the Hugo Boss page left click "Watch Now" to open in the same tab
6. Observe the pricing and resources nav loads
